### PR TITLE
Fix: Prevent issue "WinUI Desktop Window object has already been closed"

### LINKED ISSUE #15599: Bug: System.Runtime.InteropServices.COMException: The operation identifier is not valid.
Sentry Issue: [FILES-APP-E](https://files-org.sentry.io/issues/5456839101/?referrer=github_integration)

```
System.Runtime.InteropServices.COMException: The operation identifier is not valid.

The WinUI Desktop Window object has already been closed.
  ?, in async void InfoPaneViewModel.ContentPageContext_PropertyChanged(object sender, PropertyChangedEventArgs e)
```

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #15599

I'm not sure if this is the best fix, but it should at least prevent the crash.

### CODE REVIEW

**Comment by yaira2** [comment]
I'd prefer to come up with a better fix long term, but I think we should do this as a temporary solution.

File: src/Files.App/ViewModels/UserControls/InfoPaneViewModel.cs [hunk 1]
```diff
@@ -136,12 +138,27 @@ private async void ContentPageContext_PropertyChanged(object? sender, PropertyCh
...
+					{
+						// Handle exception in case WinUI Windows is closed
+						// (see https://github.com/files-community/Files/issues/15599)
+
+						SentrySdk.CaptureException(ex, scope =>
+						{
+							scope.User.Id = generalSettingsService.UserId;
+							scope.Level = SentryLevel.Warning;
+						});
+					}
```
Comment (hishitetsu) [inline src/Files.App/ViewModels/UserControls/InfoPaneViewModel.cs:160]: Shouldn't it also log to a file?
I'd like to incorporate sending reports to Sentry into the logger soon.

Comment (yaira2) [reply src/Files.App/ViewModels/UserControls/InfoPaneViewModel.cs:160]: I suppose

File: src/Files.App/ViewModels/UserControls/InfoPaneViewModel.cs [hunk 2]
```diff
@@ -136,12 +139,29 @@ private async void ContentPageContext_PropertyChanged(object? sender, PropertyCh
...
+						// Handle exception in case WinUI Windows is closed
+						// (see https://github.com/files-community/Files/issues/15599)
+
+						SentrySdk.CaptureException(ex, scope =>
+						{
+							scope.User.Id = generalSettingsService.UserId;
+							scope.Level = SentryLevel.Warning;
+						});
+
+						App.Logger.LogError(ex, ex.Message);
```
Comment (hishitetsu) [inline src/Files.App/ViewModels/UserControls/InfoPaneViewModel.cs:162]:
```suggestion
						App.Logger.LogWarning(ex, ex.Message);
```

File: src/Files.App/Views/MainPage.xaml.cs [hunk 3]
```diff
@@ -442,14 +444,30 @@ private void PaneSplitter_ManipulationCompleted(object sender, ManipulationCompl
...
+				// Handle exception in case WinUI Windows is closed
+				// (see https://github.com/files-community/Files/issues/15599)
+
+				SentrySdk.CaptureException(ex, scope =>
+				{
+					scope.User.Id = generalSettingsService.UserId;
+					scope.Level = SentryLevel.Warning;
+				});
+
+				App.Logger.LogError(ex, ex.Message);
```
Comment (hishitetsu) [inline src/Files.App/Views/MainPage.xaml.cs:469]:
```suggestion
				App.Logger.LogWarning(ex, ex.Message);
```

File: src/Files.App/Views/MainPage.xaml.cs [hunk 4]
```diff
@@ -442,14 +444,30 @@ private void PaneSplitter_ManipulationCompleted(object sender, ManipulationCompl
...
+			}
 
 			UpdatePositioning();
```
Comment (hishitetsu) [inline src/Files.App/Views/MainPage.xaml.cs:472]: Shouldn't this be included in `try`?

**Comment by 0x5bfa** [comment]
Feel free to add Sentry to GlobalUsings．

File: src/Files.App/Views/MainPage.xaml.cs (hunk 4 above)
Comment (yaira2) [reply src/Files.App/Views/MainPage.xaml.cs:472]: It shouldn't cause a crash but I guess we should only update if the window is visible.

**Pull request merged**